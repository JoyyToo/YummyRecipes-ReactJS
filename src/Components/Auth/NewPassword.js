import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import TextField from 'material-ui/TextField';
import Paper from 'material-ui/Paper';
import { notify } from 'react-notify-toast';
import axiosInstance from '../Constants/AxiosCall';
import '../styles.css';


class NewPassword extends Component {
  // initialize state
  constructor(props) {
    super(props);
    this.state = {
      newpassword: '',
    };
  }

  // handle user input
  handleInputChange = (event) => {
    const target = event.target;
    const value = target.value;
    const name = target.name;
    this.setState({ [name]: value });
  }

  // handle new password request
  handleNewPassword = () => {
    const payload = new FormData();
    payload.set('newpassword', this.state.newpassword);
    const token = this.props.match.params.token;

    // send POST request to API
    axiosInstance.post(`auth/new-password/${token}`, payload)
      .then((response) => {
        notify.show(response.data.message, 'success', 4000);
        this.props
          .history
          .push('/login');
      })
      .catch((error) => {
        if (error.response) {
          notify.show(error.response.data.message, 'error', 4000);
        }
      });
  }

  // render new password form
  render() {
    const style = {
      marginLeft: 20,
      width: 323,
    };
    const divstyle = {
      width: 430,
      marginTop: 300,
      margin: 'auto',
      textAlign: 'left',
      paddingLeft: 20,
    };
    return (
      <MuiThemeProvider>
        <div className="main">
          <div className="inputs">
            <Paper zDepth={2} style={divstyle}>
              <p className="heading">New Password Here</p>

              <TextField
                floatingLabelText="New Password"
                name="newpassword"
                value={this.state.newpassword}
                style={style}
                type="password"
                onChange={this.handleInputChange}
              /><br />

              <button
                className="network"
                id="one"
                onClick={(event => this.handleNewPassword(event))}
              >RESET
              </button>
            </Paper><br />
          </div>
        </div>
      </MuiThemeProvider>
    );
  }
}

export default NewPassword;
