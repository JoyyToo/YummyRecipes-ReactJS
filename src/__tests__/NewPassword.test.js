import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import NewPassword from '../Components/Auth/NewPassword'

describe('NewPassword component', () => {
    const wrapper = shallow(<NewPassword />);
  
    it('renders properly', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders 1 <AddCategory/> component', () =>{
        const component = shallow(<NewPassword/>);
        expect(component).toHaveLength(1);
    });

    it('contains MuiThemeProvider', () => {
        expect(wrapper.find('MuiThemeProvider')).toHaveLength(1);
    });

    it('contains divs', () => {
        expect(wrapper.find('div')).toHaveLength(2);
    });

    it('contains Paper', () => {
        expect(wrapper.find('Paper')).toHaveLength(1);
    });

    it('contains TextField ', () => {
        expect(wrapper.find('TextField')).toHaveLength(1);
    });

    it('contains button', () => {
        expect(wrapper.find('button')).toHaveLength(1);
    });

    it('contains br', () => {
        expect(wrapper.find('br')).toHaveLength(2);
    });

    it('has initial state', () => {
        expect(wrapper.state().newpassword).toEqual('');
    });

    it('renders component in MuiThemeProvider', () => {
        expect(wrapper.find('MuiThemeProvider')).toHaveLength(1);
    });

    it('renders component in divs', () => {
        expect(wrapper.find('div')).toHaveLength(2);
    });

    it('renders component in Paper', () => {
        expect(wrapper.find('Paper')).toHaveLength(1);
    });

    it('renders component in TextField ', () => {
        expect(wrapper.find('TextField')).toHaveLength(1);
    });

    it('renders component in button', () => {
        expect(wrapper.find('button')).toHaveLength(1);
    });

    it('renders component in br', () => {
        expect(wrapper.find('br')).toHaveLength(2);
    });

    it('has initial state', () => {
        expect(wrapper.state().newpassword).toEqual('');
    });


    it('has the correct form fields', () => {
        expect(wrapper.find('[name="token"]')).toHaveLength(1);
        expect(wrapper.find('[name="newpassword"]')).toHaveLength(1);
    });

})