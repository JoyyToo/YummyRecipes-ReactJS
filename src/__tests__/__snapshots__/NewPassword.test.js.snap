// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPassword component renders properly 1`] = `
<MuiThemeProvider>
  <div
    className="main"
  >
    <div
      className="inputs"
    >
      <Paper
        circle={false}
        rounded={true}
        style={
          Object {
            "margin": "auto",
<<<<<<< HEAD
<<<<<<< HEAD
            "marginTop": 300,
=======
>>>>>>> [Feature 155119223]:
=======
            "marginTop": 300,
>>>>>>> [chore] Add tests
            "paddingLeft": 20,
            "textAlign": "left",
            "width": 430,
          }
        }
        transitionEnabled={true}
        zDepth={2}
      >
<<<<<<< HEAD
<<<<<<< HEAD
=======
>>>>>>> [chore] Add tests
        <p
          className="heading"
        >
          New Password Here
        </p>
<<<<<<< HEAD
=======
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Token"
          fullWidth={false}
          multiLine={false}
          name="token"
          onChange={[Function]}
          rows={1}
          style={
            Object {
              "marginLeft": 20,
              "width": 323,
            }
          }
          type="text"
          underlineShow={true}
          value=""
        />
        <br />
>>>>>>> [Feature 155119223]:
=======
>>>>>>> [chore] Add tests
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="New Password"
          fullWidth={false}
          multiLine={false}
          name="newpassword"
          onChange={[Function]}
          rows={1}
          style={
            Object {
              "marginLeft": 20,
              "width": 323,
            }
          }
<<<<<<< HEAD
<<<<<<< HEAD
          type="password"
=======
          type="text"
>>>>>>> [Feature 155119223]:
=======
          type="password"
>>>>>>> [chore] Add tests
          underlineShow={true}
          value=""
        />
        <br />
        <button
          className="network"
          id="one"
          onClick={[Function]}
        >
          RESET
        </button>
      </Paper>
      <br />
    </div>
  </div>
</MuiThemeProvider>
`;
